### PR TITLE
Complete adding Piranha compiler and dependencies

### PR DESCRIPTION
**Pull request by wxzeng**
Task 13

### CODE REVIEW

File: include/language_rules.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#ifndef HELLO_WORLD_COMPILER_LANGUAGE_RULES_H
```
Comment (ange-yaghi) [inline include/language_rules.h]: Header guard blocks should be named after the file. For this project we use  URBANITY_<file name>_H. So in this case, we should have: #ifndef URBANITY_LANGUAGE_RULES_H   etc

File: include/language_rules.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+class LanguageRules : public piranha::LanguageRules {
+public:
+	LanguageRules();
+	~LanguageRules();
+
+	virtual void registerBuiltinNodeTypes();
+};
+
+#endif /* HELLO_WORLD_COMPILER_LANGUAGE_RULES_H */
```
Comment (ange-yaghi) [inline include/language_rules.h]: Always make sure files end in a newline. If this was in the original file then let me know so I can remove it in that repo.

File: project/urbanity/urbanity.vcxproj [hunk 3]
```diff
@@ -157,10 +160,13 @@ py $(ProjectDir)\..\..\scripts\copy_dlls.py $(PlatformTarget) $(Configuration)
       <PreprocessorDefinitions>_DEBUG;_CONSOLE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <ConformanceMode>true</ConformanceMode>
       <PrecompiledHeaderFile>pch.h</PrecompiledHeaderFile>
+      <AdditionalIncludeDirectories>C:\Development\urbanity\dependencies\piranha\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (ange-yaghi) [inline project/urbanity/urbanity.vcxproj]: All directories should be relative. It's done correctly later on in this project file but this path won't work on anyone else's computer.

File: project/urbanity/urbanity.vcxproj.filters [hunk 4]
```diff
@@ -37,6 +37,12 @@
     <Filter Include="Source Files\simulator">
       <UniqueIdentifier>{d002ebc7-fb60-4767-ad16-d81ad52c6b8f}</UniqueIdentifier>
     </Filter>
+    <Filter Include="Header Files\Piranha">
```
Comment (ange-yaghi) [inline project/urbanity/urbanity.vcxproj.filters]: All filters currently follow a lowercase only naming convention so we should consider being consistent about it

File: src/language_rules.cpp [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+	// Operations =========================================
+	registerOperator(
+		{ piranha::IrBinaryOperator::ADD, &piranha::FundamentalType::IntType, &piranha::FundamentalType::IntType },
+		"int_add"
+	);
+
+	// Conversions ========================================
+	registerConversion({ &piranha::FundamentalType::StringType, &piranha::FundamentalType::IntType }, "string_to_int");
+}
```
Comment (ange-yaghi) [inline src/language_rules.cpp]: Make sure to have a newline at the end of the file.

File: project/urbanity/urbanity.vcxproj (hunk 3 above)
Comment (wxzeng) [reply project/urbanity/urbanity.vcxproj]: How do I make it relative?

Comment (wxzeng) [reply project/urbanity/urbanity.vcxproj]: $(ProjectDir) I suppose

Comment (ange-yaghi) [reply project/urbanity/urbanity.vcxproj]: Yes exactly, something like `$(ProjectDir)/../../dependencies/piranha/include/`. Also make sure that this applies to all configurations and architectures

**Pull request merged**